Source/Core/AudioCommon/CubebUtils.cpp

```diff
@@ -8,14 +8,52 @@
 
 #include "AudioCommon/CubebUtils.h"
 #include "Common/CommonPaths.h"
-#include "Common/Logging/Log.h"
 #include "Common/Logging/LogManager.h"
 #include "Common/StringUtil.h"
 
 #include <cubeb/cubeb.h>
 
 static ptrdiff_t s_path_cutoff_point = 0;
 
+CubebUtils::ScopedThreadAccess::ScopedThreadAccess()
+{
+#ifdef _WIN32
+  // COINIT_MULTITHREADED seems to be the better choice here given we handle multithread access
+  // safely ourself. It should also offer better performance. Note that calls to this from the main
+  // thread will return RPC_E_CHANGED_MODE as some external library is calling CoInitialize
+  // with COINIT_APARTMENTTHREADED without uninitializing it. We keep going. Cubeb specifies we need
+  // COINIT_MULTITHREADED: https://github.com/kinetiknz/cubeb/issues/416 but that can't be a problem
+  result = CoInitializeEx(nullptr, COINIT_MULTITHREADED);
+  if (result != RPC_E_CHANGED_MODE && FAILED(result))
+    ERROR_LOG(AUDIO, "cubeb: Failed to initialize the COM library");
+#endif
+}
+CubebUtils::ScopedThreadAccess::~ScopedThreadAccess()
+{
+#ifdef _WIN32
+  if (SUCCEEDED(result))
+    CoUninitialize();
+#endif
+}
+
+bool CubebUtils::ScopedThreadAccess::Succeeded() const
+{
+#ifdef _WIN32
+  return result == RPC_E_CHANGED_MODE || SUCCEEDED(result);
+#else
+  return true;
+#endif
+}
+
+// Due to the way Dolphin is, we can't guarantee a cubeb context will be created and destroyed
+// from the same thread, this means that we can call CoInitialize on creation but not on destroy,
+// as we don't want to CoUninitialize a random thread, so before calling DestroyContext(),
+// we need to call CoInitialize ourselves. In case it was the same thread as the one it was created
+// on, the new init will just stack and a ScopedThreadAccess will be popped immediately,
+// if it's another thread, then the original ScopedThreadAccess will be called when the thread is
+// closed, avoiding memory leaks
+thread_local std::vector<CubebUtils::ScopedThreadAccess> scoped_thread_accesses;
```

Comment:
This should also be marked `static`, given it's only used within this translation unit. This will cause a `-Wmissing-declarations` warning otherwise. We also normally keep all static variables together near the top of the file to make for consistent reading